app/models/user.rb

```diff
@@ -24,15 +28,36 @@ def notify_slack(message)
     end
   end
 
-  after_initialize :set_default_role, if: :new_record?
-
   def name
     "#{first_name} #{last_name}"
   end
 
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0, 20]
+      user.role = :user
+
+      user.first_name = auth.info.first_name
+      user.last_name = auth.info.last_name
+    end
+  end
+
   private
 
   def set_default_role
     self.role ||= :user
   end
+
+  def set_user_id
```

Comment:
what about this user ids have to be > 4000 have i dream that or is this not the place to implement it?

Reply:
This method takes the maximum of the already existing user ids and increments it. If no maximum user id exists it it takes the configuration value `start_user_id`, which is currently set to 4000. So this is read from the config so we can change it easily later.